src/ai/backend/agent/agent.py

```diff
@@ -1301,6 +1321,7 @@ def _rm(path: Path) -> None:
                         log.debug("cleanup requested: {} ({})", body["ID"], body.get("reason"))
                         terminated_kernels[body["ID"]] = ContainerLifecycleEvent(
                             KernelId(UUID(body["ID"])),
+                            SessionId(UUID(body["SID"])),
                             ContainerId(body["CID"]),
                             LifecycleEvent.DESTROY,
                             KernelLifecycleEventReason.from_value(body.get("reason"))
```

Comment:
I think we can utilize `kernel_registry` and if matching kernel information does not exist we can also read `session-id` container label instead of updating Watcher's file format.

Reply:
I updated this part to utilize `kernel_registry`